source/production/F0.Analyzers/CodeAnalysis/CodeRefactorings/ObjectInitializer.cs

```diff
@@ -102,18 +109,15 @@ private static async Task<Document> CreateObjectInitializer(Document document, S
 			return documentEditor.GetChangedDocument();
 		}
 
-		private static IEnumerable<ISymbol> GetMutableMembers(ref TypeInfo typeInfo, Compilation compilation)
+		private static IEnumerable<ISymbol> GetMutableMembers(in TypeInfo typeInfo, Compilation compilation)
 		{
 			var members = new HashSet<ISymbol>(SymbolNameComparer.Instance);
 			var mutableMembers = new List<ISymbol>();
 
 			var type = typeInfo.Type;
-			var objectType = compilation.GetSpecialType(SpecialType.System_Object);
-			var valueType = compilation.GetSpecialType(SpecialType.System_ValueType);
-
-			while (type is { } && type != objectType && type != valueType)
+			while (type is not null && type.SpecialType is not SpecialType.System_Object and not SpecialType.System_ValueType)
```

Comment:
For better readability maybe extract 3 local variables of type boolean, which have a sensitive name?

Reply:
I personally don't want to split up the logical pattern, I'mn afraid I'm a sucker for C# 9.0 features.
However giving this condition a name is an excellent idea!

This expression evaluates to true if the current type has a BaseType, but this BaseType is neither `System.Object` (for reference types) nor `System.ValueType` (for value types) to exit the loop as soon as possible.

I was thinking about
* `IsTerminalBaseType` (similar to "terminal" LINQ methods)
* `IsFinalBaseType` (like `final` in Java, but that would actually suggest the exact opposite)
* `IsUltimateBaseType` (well, sounds like something from a _MOBA_)

Reply:
Reading the docs about [Inheritance in C# and .NET](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/inheritance) point out [Implicit inheritance](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/inheritance#implicit-inheritance).
And this expression returns `false` if the BaseType is implicitly inherited.
And to invert that into a positive expression, I am going with `HasExplicitBaseType`

Reply:
-> extracted a local function named `IsExplicitBaseType`